File: Elixir/counting_words_with_a_given_prefix.ex
# https://leetcode.com/problems/counting-words-with-a-given-prefix/
# 2185. Counting Words With a Given Prefix

defmodule Solution do
  def prefix_count(words, pref) do
    words
    |> Enum.filter(&String.starts_with?(&1, pref))
    |> Enum.count()
  end
end

File: rust/counting_words_with_a_given_prefix.rs
// https://leetcode.com/problems/counting-words-with-a-given-prefix/
// 2185. Counting Words With a Given Prefix

fn prefix_count(words: Vec<String>, pref: String) -> i32 {
        
    words
        .into_iter()
        .filter(|w| {pref.len() <= w.len() && w[..pref.len()] == pref})
        .collect::<Vec<String>>()
        .len() as i32
    
}

fn prefix_count(words: Vec<String>, pref: String) -> i32 {
    words.iter().filter(|word| word.starts_with(pref.as_str())).count() as _
}